crates/types/src/schema/metadata/updater/mod.rs

```diff
@@ -471,33 +481,37 @@ impl SchemaUpdater {
             ));
         };
 
+        // Check we don't remove services
         let mut services_to_remove = Vec::default();
-
         for service in existing_deployment.services.values() {
             if !proposed_services.contains_key(&service.name) {
                 services_to_remove.push(service.name.clone());
             }
         }
-
         if !services_to_remove.is_empty() {
             // we don't allow removing services as part of update deployment
             return Err(SchemaError::Deployment(DeploymentError::RemovedServices(
                 services_to_remove,
             )));
         }
 
-        let mut computed_services = HashMap::with_capacity(proposed_services.len());
-
         // Compute service schemas
+        let mut computed_services = HashMap::with_capacity(proposed_services.len());
         for (service_name, service) in proposed_services {
+            let previous_service_revision = existing_deployment
+                .services
+                .get(service_name.as_ref())
+                .map(|arc| arc.deref());
             let service_revision = self.create_service_revision(
                 deployment_id,
                 &deployment_metadata.ty,
                 &service_name,
                 service,
+                previous_service_revision,
                 RemovedHandlerBehavior::Fail,
                 ServiceTypeMismatchBehavior::Fail,
                 ServiceLevelSettingsBehavior::Preserve,
+                ServiceRevisionBehavior::KeepExisting,
```

Comment:
It's not clear to me why we need to keep the current revision on a schema update. You said that this is a bug and we instead should instead use the previous version, but it's not clear why to me. Any change to the schema should bump the revision, correct.

The other main question is about the mechanics where u decide if we bump or not (via the enum), why not instead set the correct version on the returned service_revision instead passing the previous revision and wither you should bump it or not.
So basically
```
let mut service_revision = self.create_service_revision(...);
service_revision.version = previous.map(|v| v + 1).unwrap_or(1);
```